x509/cert_validations.go

```diff
@@ -57,16 +57,28 @@ func validateCertChain(certChain []*x509.Certificate, expectedLeafEku x509.ExtKe
 		if signingTime != nil && (signingTime.Before(cert.NotBefore) || signingTime.After(cert.NotAfter)) {
 			return fmt.Errorf("certificate with subject %q was not valid at signing time of %s", cert.Subject, signingTime.UTC())
 		}
-
+		selfSigned, selfSignedError := isSelfSigned(cert)
 		if i == len(certChain)-1 {
-			if !isSelfSigned(cert) {
-				return errors.New("certificate chain must end with a root certificate (root certificates are self-signed)")
+			if !selfSigned {
+				if selfSignedError != nil {
```

Comment:
if it is `selfSigned` but `selfSignedError`  is not nil, do we expect to skip the error?

Reply:
> if it is `selfSigned` but `selfSignedError` is not nil, do we expect to skip the error?

This will never happen. Because when `isSelfSigned` returns true, `err` will always be `nil`: https://github.com/patrickzheng200/notation-core-go/blob/495fc78c6158d368c7a57a85d6094fe0dda5d036/x509/cert_validations.go#L107